grails-app/assets/javascripts/show.js

```diff
@@ -127,6 +127,21 @@ $(document).ready(function() {
         if(code!=""){
             $('#assertionSubmitProgress').css({'display':'block'});
 
+            if (myAnnotationQueryId) {
```

Comment:
Should you add / remove alerts even if the form submit is invalid?  What if the assertion add fails?

If the add / delete Alert works, shouldn't you update $('#notifyChangeCheckbox').prop('data-origstate') in case the form is resubmitted after a validation error, otherwise you might have a duplicate add / delete call?

Reply:
add/delete alert is now moved into success callback of add assertion. So only when an assertion is added successfully an add/delete will be called.